src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -175,8 +176,9 @@ public int[] getRequiredTokens() {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final DetailAST leftCurly = findLeftCurly(ast);
-        if (leftCurly != null) {
+        final Optional<DetailAST> lcurlyOptional = findLeftCurly(ast);
```

Comment:
Please never name Optional variables as `Optional...`. We force the developer to think about the semantics of this type by contract.

```suggestion
        final Optional<DetailAST> leftCurly = findLeftCurly(ast);
```

Reply:
done. renamed optional to leftCurly and the value we get from the Optional to leftCurlyAST is it good ?

Reply:
yep